powerstore/resource_smb_share.go

```diff
@@ -0,0 +1,658 @@
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"terraform-provider-powerstore/client"
+	"terraform-provider-powerstore/models"
+)
+
+// newSMBShareResource returns SMBShare new resource instance
+func newSMBShareResource() resource.Resource {
+	return &resourceSMBShare{}
+}
+
+type resourceSMBShare struct {
+	client *client.Client
+}
+
+// Metadata defines resource interface Metadata method
+func (r *resourceSMBShare) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_smb_share"
+}
+
+// Schema defines resource interface Schema method
+func (r *resourceSMBShare) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+
+		MarkdownDescription: "This resource is used to manage the smb share entity of PowerStore Array. We can Create, Update and Delete the smb share using this resource. We can also import an existing smb share from PowerStore array.",
+		Description:         "This resource is used to manage the smb share entity of PowerStore Array. We can Create, Update and Delete the smb share using this resource. We can also import an existing smb share from PowerStore array.",
+
+		Attributes: SMBShareSchema(),
+	}
+}
+
+// SMBShareSchema defines resource interface Schema method
+func SMBShareSchema() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			Computed:            true,
+			Description:         "The unique identifier of the SMB Share.",
+			MarkdownDescription: "The unique identifier of the SMB Share.",
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"file_system_id": schema.StringAttribute{
+			MarkdownDescription: "The unique identifier of the file system on which the SMB Share is created.",
+			Description:         "The unique identifier of the file system on which the SMB Share is created.",
+			Required:            true,
+			Validators: []validator.String{
+				stringvalidator.LengthAtLeast(1),
+			},
+		},
+		"name": schema.StringAttribute{
+			MarkdownDescription: "The name of the SMB Share.",
+			Description:         "The name of the SMB Share.",
+			Required:            true,
+			Validators: []validator.String{
+				stringvalidator.LengthBetween(1, 80),
+			},
+		},
+		"path": schema.StringAttribute{
+			MarkdownDescription: "The local path to export relative to the NAS Server.",
+			Description:         "The local path to export relative to the NAS Server.",
+			Required:            true,
+			Validators: []validator.String{
+				stringvalidator.LengthAtLeast(1),
+			},
+		},
+		"description": schema.StringAttribute{
+			MarkdownDescription: "User defined SMB share description.",
+			Description:         "User defined SMB share description.",
+			Optional:            true,
+			Computed:            true,
+			Validators: []validator.String{
```

Comment:
Need to support empty string for removal of description?

Reply:
updated